practice-two/src/stores/reducers.js

```diff
@@ -0,0 +1,28 @@
+export const initialCustomer = {
+  customers: [],
+};
+
+export const customerReducer = (state, action) => {
+  switch (action.type) {
+    case 'CREATE_CUSTOMER': {
+      return { customers: [...state.customers, action.payload], ...state };
+    }
+    case 'UPDATE_CUSTOMER': {
+      const updatedCustomer = state.customers.map((customer) => {
+        if (customer.id === action.payload.id) {
+          return { ...customer, ...action.payload };
+        }
+        return customer;
+      });
+      return { ...state, customers: updatedCustomer };
+    }
+    case 'DELETE_CUSTOMER': {
+      const filteredCustomers = state.customers.filter(
+        (customer) => customer.id !== action.payload
```

Comment:
check for payload or payload.id

Reply:
Sorry, it is absolutely a payload.id
![image](https://github.com/datdragon777/React-Training/assets/74806826/221ad146-6df4-42d9-8f06-615276895368)